packages/config/src/handlers/forms/validation.ts

```diff
@@ -0,0 +1,424 @@
+              REQUIRED_FIELDS_IN_SECTION[sec.id] ?? []
+            ).filter((reqField) => !fieldsInSection.includes(reqField))
+            return {
+              message: `Missing required fields "${notIncludedRequiredFields.join(
+                ', '
+              )}" found`
+            }
+          }
+        )
+        .refine(
+          (sec) => {
+            if (!REQUIRED_FIELDS_IN_SECTION[sec.id]) {
+              return true
+            }
+            const nonCustomfieldsInSection = sec.groups.flatMap((group) =>
+              group.fields
+                .filter(({ custom }) => !Boolean(custom))
+                .filter(({ type }) => type !== 'DIVIDER')
+                .filter(
+                  ({ name }) =>
+                    !(REQUIRED_FIELDS_IN_SECTION[sec.id] ?? []).includes(name)
+                )
+                .map(({ name }) => name)
+            )
+            return (
+              nonCustomfieldsInSection.filter(
+                (nonCustomField) =>
+                  !(OPTIONAL_FIELDS_IN_SECTION[sec.id] ?? []).includes(
+                    nonCustomField
+                  )
+              ).length === 0
+            )
+          },
+          (sec) => {
+            const nonCustomfieldsInSection = sec.groups.flatMap((group) =>
+              group.fields
+                .filter(({ custom }) => !Boolean(custom))
+                .filter(({ type }) => type !== 'DIVIDER')
+                .filter(
+                  ({ name }) =>
+                    !(REQUIRED_FIELDS_IN_SECTION[sec.id] ?? []).includes(name)
+                )
+                .map(({ name }) => name)
+            )
+            const unrecognizedFields = nonCustomfieldsInSection.filter(
+              (nonCustomField) =>
+                !(OPTIONAL_FIELDS_IN_SECTION[sec.id] ?? []).includes(
+                  nonCustomField
+                )
+            )
+            return {
+              message: `Use "custom" property to make a field custom. Unrecognized non custom fields "${unrecognizedFields.join(
+                ', '
+              )}" found`
+            }
+          }
+        )
+    )
+    .refine(
+      (sections) =>
+        sections.filter(({ id }) =>
+          REQUIRED_SECTIONS.includes(id as typeof REQUIRED_SECTIONS[number])
+        ).length >= 2,
+      {
+        message: `${REQUIRED_SECTIONS.map((sec) => `"${sec}"`).join(
+          ' & '
+        )} sections are required`
+      }
+    )
+    .refine(
+      (sections) => {
+        findDuplicates(
+          sections
+            .flatMap((sec) => [
+              ...(sec.mapping?.template ?? []).map(
+                ({ fieldName }) => fieldName
+              ),
+              ...sec.groups
+                .flatMap((group) => group.fields)
+                .map(({ mapping }) => mapping?.template?.fieldName)
+            ])
+            .filter((maybeName): maybeName is string => Boolean(maybeName))
+        ).length === 0
+      },
```

Comment:
```suggestion
      (sections) =>
        findDuplicates(
          sections
            .flatMap((sec) => [
              ...(sec.mapping?.template ?? []).map(
                ({ fieldName }) => fieldName
              ),
              ...sec.groups
                .flatMap((group) => group.fields)
                .map(({ mapping }) => mapping?.template?.fieldName)
            ])
            .filter((maybeName): maybeName is string => Boolean(maybeName))
        ).length === 0,
```
@euanmillar Sorry my bad. I wasn't returning the predicate

Reply:
Works perfectly @Zangetsu101 .  I applied the fix in my PR.